spec/compiler/semantic/enum_spec.cr

```diff
@@ -436,4 +436,16 @@ describe "Semantic: enum" do
     method = result.program.types["Foo"].lookup_first_def("bar", block: false).not_nil!
     method.always_inline?.should be_true
   end
+
+  it "errors if defining initialize in Enum (#7238)" do
+    assert_error %(
+      enum Foo
+        FOO = 1
+
+        def initialize
+        end
+      end
+      ),
+      "enums can't define an `initialize` method, try using `def self.new`"
```

Comment:
This advise is not good, because `def self.new` doesn't work either:
```cr
enum Foo
  FOO = 1
  
  def self.new(value)
    puts "self.new caleld"
    new(value)
  end
end

Foo.new(1)
```